crypto/helpers.go

```diff
@@ -15,19 +15,40 @@ import (
 	"fmt"
 	"golang.org/x/crypto/pbkdf2"
 	"io"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"math/big"
+	"errors"
 )
 
 // https://www.socketloop.com/tutorials/golang-padding-un-padding-data
 // https://www.socketloop.com/tutorials/golang-example-for-rsa-package-functions-example
 
-func RandomBytes(size int) (randomBytes []byte) {
-	randomBytes = make([]byte, size)
+type KeyType string
+const (
+	KeyTypeRSA KeyType = "rsa"
+	KeyTypeEC KeyType = "ec"
+)
+
+func RandomBytes(size int) ([]byte, error) {
+	randomBytes := make([]byte, size)
 	_, err := rand.Read(randomBytes)
 	if err != nil {
-		panic(fmt.Errorf("Could not generate random bytes: %s", err.Error()))
-	} else {
-		return randomBytes
+		return nil, fmt.Errorf("Could not generate random bytes: %s", err)
 	}
+
+	return randomBytes, nil
+}
+
+func RandomIntBetween(x, y *big.Int) (*big.Int, error) {
+	dupe := new(big.Int)
+	dupe.SetString(y.String(), 10)
```

Comment:
Is all the string stuff necessary? Wouldn't something like this work?

	diff := new(big.Int)
	diff.Sub(y, x)
	r, _ := rand.Int(rand.Reader, diff)
        return r.Add(r, x)